sdk/src/types/block/capabilities.rs

```diff
@@ -60,30 +91,28 @@ impl<Flag: CapabilityFlag> Capabilities<Flag> {
         self
     }
 
-    /// Disables every possible flag.
-    pub fn set_none(&mut self) -> &mut Self {
-        *self = Default::default();
-        self
-    }
-
-    /// Enables a given flag.
-    pub fn add_capability(&mut self, flag: Flag) -> &mut Self {
+    fn inner_add_capability(&mut self, flag: Flag, trim: bool) -> &mut Self {
         if self.bytes.len() <= flag.index() {
             let mut v = Box::<[_]>::from(self.bytes.clone()).into_vec();
             v.resize(flag.index() + 1, 0);
             // Unwrap: safe because the indexes are within u8 bounds
             self.bytes = v.into_boxed_slice().try_into().unwrap();
         }
         self.bytes[flag.index()] |= flag.as_byte();
-        self
+        if trim { self.trim() } else { self }
+    }
+
+    /// Enables a given flag.
+    pub fn add_capability(&mut self, flag: Flag) -> &mut Self {
+        self.inner_add_capability(flag, true)
     }
 
     /// Enables a given set of flags.
     pub fn add_capabilities(&mut self, flags: impl IntoIterator<Item = Flag>) -> &mut Self {
         for flag in flags {
-            self.add_capability(flag);
+            self.inner_add_capability(flag, false);
         }
-        self
+        self.trim()
```

Comment:
Do we actually need to trim? Adding capabilities should never end up adding an empty byte? Or am I missing something?

Reply:
🤔 I think you're right